src/test/java/net/starschema/clouddb/jdbc/BQForwardOnlyResultSetFunctionTest.java

```diff
@@ -831,20 +798,22 @@ private void mockResponse(String jsonResponse) throws Exception {
   }
 
   @Test
-  public void testStatelessQuery() throws SQLException {
-    NewConnection("&useLegacySql=false&jobcreationmode=JOB_CREATION_OPTIONAL");
-    StatelessQuery.assumeStatelessQueriesEnabled(
-        BQForwardOnlyResultSetFunctionTest.con.getCatalog());
-    final Statement stmt =
-        BQForwardOnlyResultSetFunctionTest.con.createStatement(
-            ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY);
-    final ResultSet result = stmt.executeQuery(StatelessQuery.exampleQuery());
-    final String[][] rows = BQSupportMethods.GetQueryResult(result);
-    Assertions.assertThat(rows).isEqualTo(StatelessQuery.exampleValues());
-
-    final BQForwardOnlyResultSet bqResultSet = (BQForwardOnlyResultSet) result;
-    Assertions.assertThat(bqResultSet.getJobId()).isNull();
-    Assertions.assertThat(bqResultSet.getQueryId()).contains("!");
+  public void testStatelessQuery() throws SQLException, IOException {
+    try (Connection statelessConnection =
+        connect("&useLegacySql=false&jobcreationmode=JOB_CREATION_OPTIONAL")) {
```

Comment:
maybe there should be a ConnectionBuilder that removes all this JDBC Url crap from our own tests ?
Seems like an obvious way to cause bugs.